packages/cli-kit/src/public/node/toml.ts

```diff
@@ -1,17 +1,34 @@
 import {JsonMap} from '../../private/common/json.js'
 import * as toml from '@iarna/toml'
+import {Liquid} from 'liquidjs';
+import {readAndParseDotEnv } from './dot-env.js'
+import {fileExists} from './fs.js'
+import {joinPath, cwd} from './path.js'
 
 export type JsonMapType = JsonMap
 
 /**
  * Given a TOML string, it returns a JSON object.
+ * Before parsing as TOML the string is passed through a
+ * Liquid renderer with access to environment variables.
  *
  * @param input - TOML string.
  * @returns JSON object.
  */
-export function decodeToml(input: string): object {
+export async function decodeToml(input: string): object {
+  const engine = new Liquid()
   const normalizedInput = input.replace(/\r\n$/g, '\n')
   return toml.parse(normalizedInput)
+
+  let envData = process.env;
+  const envFile = joinPath(cwd(), '.env');
```

Comment:
There are conventions for multiple `.env` files, see relevant information in Vite docs: https://vitejs.dev/guide/env-and-mode#env-files